modules/database/test/std/rec/boTest.c

```diff
@@ -0,0 +1,143 @@
+/*************************************************************************\
+* Copyright (c) 2023 Karl Vestin
+* EPICS BASE is distributed subject to a Software License Agreement found
+* in file LICENSE that is included with this distribution.
+\*************************************************************************/
+
+#include <unistd.h>
+
+#include "dbUnitTest.h"
+#include "testMain.h"
+#include "errlog.h"
+#include "dbAccess.h"
+#include "menuAlarmSevr.h"
+#include "menuIvoa.h"
+
+#include "boRecord.h"
+
+void recTestIoc_registerRecordDeviceDriver(struct dbBase *);
+
+static void test_soft_output(void){
+    /* set soft channel */
+    testdbPutFieldOk("test_bo_rec.DTYP", DBF_STRING, "Soft Channel");
+    testdbPutFieldOk("test_bo_rec.OUT", DBF_STRING, "test_bo_link_rec");
+
+    /* set VAL to process record */
+    testdbPutFieldOk("test_bo_rec.VAL", DBF_SHORT, TRUE);
+
+    /* verify that OUT record is updated */
+    testdbGetFieldEqual("test_bo_link_rec.VAL", DBF_SHORT, TRUE);
+
+    // number of tests = 4
+}
+
+static void test_high(void){
+    const int high_time = 2;
+    testMonitor* test_mon = NULL;
+    epicsTimeStamp startTime;
+    epicsTimeStamp endTime;
+    double diffTime = 0;
+    double diffTimeTolerance = 0.1;
+
+    /* set soft channel */
+    testdbPutFieldOk("test_bo_rec.DTYP", DBF_STRING, "Soft Channel");
+    testdbPutFieldOk("test_bo_rec.OUT", DBF_STRING, "test_bo_link_rec");
+
+    /* set HIGH to 2 seconds */
+    testdbPutFieldOk("test_bo_rec.HIGH", DBF_SHORT, high_time);
+
+    /* Create test monitor */
+    test_mon = testMonitorCreate("test_bo_rec.VAL", DBR_SHORT, 0);
+
+    /* Get start time */
+    epicsTimeGetCurrent(&startTime);
+
+    /* set VAL to process record */
+    testdbPutFieldOk("test_bo_rec.VAL", DBF_SHORT, TRUE);
+
+    /* wait and verfiy time */
+    testMonitorWait(test_mon);
+    epicsTimeGetCurrent(&endTime);
+
+    /* EPICS timers have a tendency to trip slightly early, hence the test tolerance is added here to avoid false positives in testing */
+    diffTime = epicsTimeDiffInSeconds(&endTime, &startTime) + diffTimeTolerance;
+    testOk(diffTime >= high_time, "HIGH time %lf", diffTime);
+
+    /* verfiy that both records are set back to 0 */
+    testdbGetFieldEqual("test_bo_rec.VAL", DBF_SHORT, FALSE);
+    testdbGetFieldEqual("test_bo_link_rec.VAL", DBF_SHORT, FALSE);
+
+    testMonitorDestroy(test_mon);
+    // number of tests = 7
+}
+
+static void test_operator_display(void){
+    /* set operator display parameters */
+    testdbPutFieldOk("test_bo_rec.ZNAM", DBF_STRING, "ZNAM_TEST");
+    testdbPutFieldOk("test_bo_rec.ONAM", DBF_STRING, "ONAM_TEST");
+    testdbPutFieldOk("test_bo_rec.DESC", DBF_STRING, "DESC_TEST");
+
+    /* verify operator display parameters */
+    testdbGetFieldEqual("test_bo_rec.ZNAM", DBF_STRING, "ZNAM_TEST");   
+    testdbGetFieldEqual("test_bo_rec.ONAM", DBF_STRING, "ONAM_TEST"); 
```

Comment:
Same as comments for `bi` record. Could implicitly test the enum conversion here.

Reply:
I agree with @henrique-silva about both of these.

Reply:
Resolved in https://github.com/epics-base/epics-base/pull/340/commits/28ecf0af3c3f2585f1571215c8b16a15ccbcab37